composite/dns-record-check/src/adapter.ts

```diff
@@ -0,0 +1,34 @@
+import { Execute, DNSResponseAnswer, AdapterRequest, DNSResponse } from '@chainlink/types'
+import { Validator } from '@chainlink/external-adapter'
+import DNS from '@chainlink/dns-query'
+import { Requester } from '@chainlink/external-adapter'
+
+const inputParams = {
+  record: true,
+  ...DNS.inputParams,
+}
+
+export const execute: Execute = async (input) => {
+  const validator = new Validator(input, inputParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const { record, ...dnsInputParams } = validator.validated.data
+
+  const dnsRequest: AdapterRequest = {
+    id: jobRunID,
+    data: dnsInputParams,
+  }
+  const dnsResponse = await DNS.execute(dnsRequest)
```

Comment:
If all input parameters are flat, which here is the case, you can just delegate the input to the underlying adapter.

```suggestion
  const dnsResponse = await DNS.execute(input)
```

Reply:
When other changes are made to this PR, in the end, it should look like something like this:

```suggestion
  const dnsQueryExecute = DNS.makeExecute(config)
  const dnsResponse = await dnsQueryExecute(input)
```

In the future with a more fluid API definition, that could transition to something like this:

```javascript
const resp = await dnsQuery.make(config).execute(input)
```